tools/cicd/integration-tests.yaml

```diff
@@ -26,13 +26,32 @@ steps:
     - PYTHON_VERSION=3.11
     - .
 
-  # Run integration tests using the built Python 3.11 image
+   # Access and store the secret in a file on a shared volume
+  - name: 'gcr.io/cloud-builders/gcloud'
+    id: 'download-secrets'
+    entrypoint: 'bash'
+    args:
+    - '-c'
+    - |
+      gcloud secrets versions access latest --secret=integration-tests --project=464139560241 > /root/secrets.sh
+    volumes:
+    - name: 'root'
+      path: /root
+
+
+  # Source the secrets and run integration tests using the built Python 3.11 image
   - name: gcr.io/$PROJECT_ID/hatch:3.11
     id: integration-tests
+    entrypoint: 'bash'
     args:
-    - run
-    - integration-test
-    waitFor: ['build-python-image']
+    - '-c'
+    - |
+      source /root/secrets.sh
+      run integration-test
```

Comment:
You will need to use `hatch run integration-test` given the entrypoint is now `bash` and not `hatch` like before.